### PR TITLE
CPR-732: Twitter Social Share

### PR DESCRIPTION
**Pull request by slothluvchunk**
Fixing twitter social share items to use the Web Intent API, allowing tweet shares to be edited by the user.

### CODE REVIEW

File: inc/wp-components/classes/class-pagination.php [hunk 1]
```diff
@@ -42,18 +42,14 @@ public function default_config() : array {
...
 
 		// Convert each HTML link to a Pagination_Item.
 		if ( ! empty( $pagination_links ) ) {
-
 			foreach ( $pagination_links as $link_html ) {
 				$this->append_child(
-
-					// Create a new pagination item using anchor HTML, and remove
-					// various url params.
+					// Create a new pagination item using anchor HTML, and remove// various url params.
```
Comment (isGabe) [inline inc/wp-components/classes/class-pagination.php]: Was this change intended as part of this pull request? If so, should probably remove the `\\`

File: inc/wp-components/classes/class-social-sharing.php [hunk 2]
```diff
@@ -83,7 +83,7 @@ public function get_twitter_component() : Social_Item {
 							'text' => $this->get_title(),
 							'url'  => $this->get_url(),
 						],
-						'https://twitter.com/share/'
+						'https://twitter.com/intent/tweet/'
```
Comment (isGabe) [inline inc/wp-components/classes/class-social-sharing.php]: When I click one a `https://twitter.com/share/` link as they are currently, it redirects to a  `https://twitter.com/intent/tweet/` URL. But this `intent/tweet` URL is the one that twitter suggests in [their docs](https://developer.twitter.com/en/docs/twitter-for-websites/tweet-button/overview), so it's probably a good idea to change it.

File: inc/wp-components/classes/class-pagination.php (hunk 1 above)
Comment (slothluvchunk) [reply inc/wp-components/classes/class-pagination.php]: Thanks - I'll remove it. I was fixing PHPCS issues - but missed that comment tag.

**Comment by slothluvchunk** [comment]
@isGabe  - conflicts resolved, can you merge this?
